### PR TITLE
Feature/language provider

### PR DESCRIPTION
**Pull request by hachiojidev**
### Remove translation id and fallback removed
Translations no longer use the translation id or fallback

```
TranslateString(200, 'My Translation')
TranslateString(201, `${variable} times`, { num: variable })
```
...becomes...
```
t'My Translation')
t('%num% times', { num: variable ))
```

### Translation source

Translations will now be from Crowdin GH integration. The master file is located at [src/config/localization/translations.json](https://github.com/pancakeswap/pancake-frontend/blob/bdfe0c36469d7652ebc6c06c4ede19f009831e3b/src/config/localization/translations.json). Anytime a new translation is added to this file Crowdin will automatically add it to the project. Then a PR will be submitted to update the generated locales at `/public/locales/[locale].json`.

Note: The master file is kept inside the `src` directory so we can include it in the bundle. This allows us to have English ready to go without an extra fetch.

### CODE REVIEW

File: src/contexts/Localization/Provider.tsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+      currentLanguage: languages[codeFromStorage],
+    }
+  })
+  const { currentLanguage } = state
+
+  useEffect(() => {
+    const fetchInitialLocales = async () => {
+      const codeFromStorage = getLanguageCodeFromLS()
+
+      if (codeFromStorage !== EN.code) {
```
Comment (hachiojidev) [inline src/contexts/Localization/Provider.tsx:33]: We only need to fetch an additional locale if the locale is not English.

File: src/contexts/Localization/Provider.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        ...prevState,
+        isFetching: false,
+      }))
+    }
+
+    fetchInitialLocales()
+  }, [setState])
+
+  const setLanguage = async (language: Language) => {
+    if (!languageMap.has(language.code)) {
```
Comment (hachiojidev) [inline src/contexts/Localization/Provider.tsx:49]: Fetch the new locale

File: src/contexts/Localization/Provider.tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+      // Merge the EN locale to ensure that any locale fetched has all the keys
+      languageMap.set(language.code, { ...enLocale, ...locale })
+      localStorage.setItem(LS_KEY, language.code)
+
+      setState((prevState) => ({
+        ...prevState,
+        isFetching: false,
+        currentLanguage: language,
+      }))
+    } else {
```
Comment (hachiojidev) [inline src/contexts/Localization/Provider.tsx:67]: Or, if we have it already just use it.

File: src/contexts/Localization/Provider.tsx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        return key
+      }
+
+      const includesVariable = translatedText.includes('%')
+
+      if (includesVariable) {
+        let interpolatedText = translatedText
+
+        Object.keys(data).forEach((dataKey) => {
+          const templateKey = new RegExp(`%${dataKey}%`, 'g')
```
Comment (ChefHutch) [inline src/contexts/Localization/Provider.tsx:93]: Sort of just rubber ducking here, but I think this'll be fine even in the scenario where we have text that includes a %, which isn't referencing a variable?

i.e. Translation: 
> Auto-Compound Bounty: %num%% of all Auto CAKE pool users’ pending yield

Code:
```
t(xxx, `Auto-Compound Bounty: ${callFee}% of all Auto CAKE pool users’ pending yield``, { num: callFee })
```

File: src/contexts/Localization/Provider.tsx [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+  })
+  const { currentLanguage } = state
+
+  useEffect(() => {
+    const fetchInitialLocales = async () => {
+      const codeFromStorage = getLanguageCodeFromLS()
+
+      if (codeFromStorage !== EN.code) {
+        const enLocale = languageMap.get(EN.code)
+        const currentLocale = await fetchLocale(codeFromStorage)
```
Comment (ChefHutch) [inline src/contexts/Localization/Provider.tsx:35]: Is a 'locale' here describing a json of all translations for that specific language?

File: src/contexts/Localization/helpers.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import { EN, LanguageCode } from '../../config/localization/languages'
+
+const publicUrl = process.env.PUBLIC_URL
+
+export const LS_KEY = 'pancakeswap_lang'
+
+export const fetchLocale = async (code: LanguageCode) => {
+  const response = await fetch(`${publicUrl}/locales/${code}.json`)
```
Comment (ChefHutch) [inline src/contexts/Localization/helpers.ts:8]: I'm still slightly confused as to how `translations.json` is populated and maintained, and then where the per-language translations are stored? Will we also have `.json` files for each language in the codebase, or are the non-EN translations fetched remotely?

**Review by ChefHutch — APPROVED** [review]
All looks great man.

Also - the lightweight docs are a nice touch.

I have more questions than CR feedback, so will give this my blessing and wishing you god speed on Monday 👍

File: src/contexts/Localization/Provider.tsx (hunk 4 above)
Comment (hachiojidev) [reply src/contexts/Localization/Provider.tsx:93]: Good eye - I have confirmed that it works.

File: src/contexts/Localization/helpers.ts (hunk 6 above)
Comment (hachiojidev) [reply src/contexts/Localization/helpers.ts:8]: Magic Hutch. Magic.

Comment (hachiojidev) [reply src/contexts/Localization/helpers.ts:8]: Once the integration is complete it will all become clear.

**Review by RabbitDoge — APPROVED** [review]

**Pull request merged**